File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

bool compare(const char * c, int index, const char * s, size_t size){
    bool result = true;
    for(int i=0;i<size;i++){
        if(c[index + i] == s[i]){}
        else{result = false;break;}
    }
    return result;
}

bool possible(const char * c){
    int len=0;
    int i=0;
    bool result = true;
    int a = 0;
    while(c[len]){len++;}
    while(i<len){
        
        if(compare(c,i,"aya",3)){
            if(a == 1){result = false;break;}
            i += 3;
            a = 1;
        }
        else if(compare(c,i,"woo",3)){
            if(a == 3){result = false;break;}
            i += 3;
            a = 3;
        }
        else if(compare(c,i,"ye",2)){
            if(a == 2){result = false;break;}
            i += 2;
            a = 2;
        }
        else if(compare(c,i,"ma",2)){
            if(a == 4){result = false;break;}
            i += 2;
            a = 4;
        }
        else{
            result = false;break;
        }
    }
    
    
    return result;
}

int solution(const char* babbling[], size_t babbling_len) {
    int answer = 0;
    for(int i=0;i<babbling_len;i++){
        if(possible(babbling[i])){
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/1/133499.옹알이 （2）-Python/옹알이 （2）.py
def solution(babbling):
    answer = 0
    for b in babbling:

        check = True
        pre = ""
        while len(b) > 0:
            if b[:3] == "aya" and pre != "aya":
                b = b[3:]
                pre = "aya"
            elif b[:2] == "ye" and pre != "ye":
                b = b[2:]
                pre = "ye"
            elif b[:3] == "woo" and pre != "woo":
                b = b[3:]
                pre = "woo"
            elif b[:2] == "ma" and pre != "ma":
                b = b[2:]
                pre = "ma"
            else:
                b = ""
                check = False
        if check:
            answer += 1

    return answer
